docs/guide/custom-ui.md

```diff
@@ -0,0 +1,207 @@
+
+As for the actual Vue code, we would place that in `modules/announcement/ui/apos/components/AnnouncementManager.vue`.
+
+Of course there are other components that can be overridden in this way, and the list is growing over time. Here are the components that can currently be overridden through configuration:
+
+| Module                     | Option             | Default          |
+| -------------------------- | ------------------ | ---------------- |
+| `@apostrophecms/piece-type`  | `components.managerModal` | `AposDocsManager`  |
+| `@apostrophecms/piece-type`  | `components.editorModal`  | `AposDocEditor`    |
+| `@apostrophecms/page`        | `components.managerModal` | `AposPagesManager` |
+| `@apostrophecms/page`        | `components.editorModal`  | `AposDocEditor`    |
+| `@apostrophecms/widget-type` | `components.widgetEditor` | `AposWidgetEditor` |
+| `@apostrophecms/widget-type` | `components.widget`       | `AposWidget`       |
+
+For readability's sake, a `.` is used in the table above to separate sub-properties of `options` (see the example above for what the actual configuration looks like). If an option exists for `@apostrophecms/piece-type` it can be used for any module that extends it.
+
+::: note
+If an option ends in `Modal`, our component is expected to embed the `AposModal` component so that the admin UI can await it with `apos.modal.execute`. For examples, look at the source code of the default components listed above.
+
+The `AposWidgetEditor` component already provides a modal dialog box in which to edit the schema of any widget, so we won't need to configure a replacement unless we want to support editing directly on the page. `AposRichTextWidgetEditor` is an example of how to do this.
+
+The `AposWidget` component has **nothing to do with a typical site visitor experience.** It is used only when displaying our widget while the page is in edit mode. While overriding this component is rare, but the `@apostrophecms/rich-text-widget` module does so to provide a "click the text to edit" experience for rich text widgets. If you're just trying to enhance your widget with frontend JavaScript, you should write a [widget player](custom-widgets.md#client-side-javascript-for-widgets) instead.
+:::
+
+## Registering custom field types
+
+Apostrophe's [schema field types](content-schema.md) cover many situations, but we might wish to add a new one.
+
+A schema field has two parts: a server-side part and a browser-side part. The server-side part is responsible for sanitizing the input received from the browser, while the browser-side part is responsible for providing the admin UI.
+
+### Implementing the server-side part
+
+Any module can register a schema field type on the server side, like this one, which provides a range field with two inputs, one for the low end of the range and one for the high end.
```

Comment:
"provides" to "creates"

Reply:
I don't understand the low end/high end part of this. It would also help to include an initial description that this will be a 1-5 rating field to help people contextualize the next parts.

Reply:
Oops, that was a leftover.